### PR TITLE
Improve Scenario Outline validation

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- add utility to convert syn errors into TokenStream
- validate example tables and check for duplicate headers
- iterate over all example blocks
- add compile-fail tests for invalid examples

## Testing
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_68895cdee05c8322b6be69f6ce28c2fc

## Summary by Sourcery

Standardize macro error conversion and enhance Scenario Outline example validation with comprehensive table structure and header consistency checks, and add compile-fail tests for invalid outline scenarios

Enhancements:
- Introduce error_to_tokens and cell_to_lit utilities to unify macro error reporting and literal generation
- Enhance Scenario Outline parsing by validating example table presence, non-empty rows, consistent column counts, uniform headers across multiple example blocks, and duplicate header detection
- Replace direct to_compile_error calls with error_to_tokens throughout macro processing

Tests:
- Add compile-fail trybuild tests and feature fixtures for missing, empty, and malformed example tables

### CODE REVIEW

File: crates/rstest-bdd-macros/src/lib.rs [hunk 1]
```diff
@@ -294,37 +302,76 @@ struct ScenarioData {
 }
 
 fn extract_examples(scenario: &gherkin::Scenario) -> Result<Option<ExampleTable>, TokenStream> {
-    if !scenario.keyword.contains("Outline") && scenario.examples.is_empty() {
+    if scenario.keyword != "Scenario Outline" && scenario.examples.is_empty() {
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/src/lib.rs:305]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: crates/rstest-bdd-macros/src/lib.rs [hunk 2]
```diff
@@ -294,36 +301,79 @@ struct ScenarioData {
...
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/src/lib.rs:304]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: crates/rstest-bdd-macros/tests/ui/outline_empty_examples.rs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+use rstest_bdd_macros::scenario;
+
+#[scenario(path = "../../../../crates/rstest-bdd-macros/tests/features/outline_empty_examples.feature")]
+fn compile_fail() {}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/tests/ui/outline_empty_examples.rs]: Has this now been resolved?

File: crates/rstest-bdd-macros/tests/ui/outline_missing_examples.rs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+use rstest_bdd_macros::scenario;
+
+#[scenario(path = "../../../../crates/rstest-bdd-macros/tests/features/outline_missing_examples.feature")]
+fn compile_fail() {}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/tests/ui/outline_missing_examples.rs]: Has this now been resolved?

File: crates/rstest-bdd-macros/src/lib.rs [hunk 5]
```diff
@@ -294,37 +302,76 @@ struct ScenarioData {
 }
 
 fn extract_examples(scenario: &gherkin::Scenario) -> Result<Option<ExampleTable>, TokenStream> {
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/src/lib.rs:330]: Has this now been resolved?

File: crates/rstest-bdd-macros/tests/ui/outline_duplicate_headers.rs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+// This test ensures duplicate headers in the Examples table produce an error.
+use rstest_bdd_macros::scenario;
+
+#[scenario(path = "../../../../crates/rstest-bdd-macros/tests/features/outline_duplicate_headers.feature")]
+fn compile_fail_duplicate_headers() {}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/tests/ui/outline_duplicate_headers.rs:5]: Has this now been resolved?

File: crates/rstest-bdd-macros/tests/ui/outline_empty_examples.rs [hunk 7]
```diff
@@ -0,0 +1,8 @@
+//! Compile-fail test for empty Examples table validation.
+use rstest_bdd_macros::scenario;
+
+/// This test fails because the Examples block only contains headers.
+#[scenario(path = "../../../../crates/rstest-bdd-macros/tests/features/outline_empty_examples.feature")]
+fn compile_fail_empty_examples() {}
+
+fn main() {}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/tests/ui/outline_empty_examples.rs:8]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Is this still the case?

- **Limited placeholder parser.** Nested or escaped braces are not supported
  in step patterns. Placeholders must be well-formed and non-overlapping.

File: crates/rstest-bdd-macros/src/lib.rs [hunk 8]
```diff
@@ -1,9 +1,8 @@
-//! Procedural macros for rstest-bdd.
+//! Attribute macros enabling Behaviour-Driven testing with `rstest`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/src/lib.rs:1]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-macros/src/lib.rs:1]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

Comment (leynos) [reply crates/rstest-bdd-macros/src/lib.rs:1]: @coderabbitai new GitHub issue, please
